tools/godotcpp.py

```diff
@@ -326,6 +326,14 @@ def options(opts, env):
             ("none", "custom", "debug", "speed", "speed_trace", "size"),
         )
     )
+    opts.Add(
+        EnumVariable(
+            "lto",
+            "Link-time optimization",
+            "auto",
```

Comment:
> The default should not be auto but none (consistent with upstream godot). 

This is false: https://github.com/godotengine/godot/blob/e4e024ab88efe74677769395886bc1b09eccbac7/SConstruct#L615C13-L615C46

Godot's LTO parameter itself may default to none (relating a few lines above), but the way they evaluate the parameter makes it actually default to "auto" on prod, and "none" elsewhere. In that, i'm mirroring what they do, not what they say they do. The main difference is that passing "auto" will result in "none" in my implementation on dev and "full" in Godot. I don't think that makes sense because you wouldn't want LTO in a dev build - hence why the default of not passing the value defaults to "none" in dev builds.

Reply:
> but the way they evaluate the parameter makes it actually default to "auto" on prod (and thus 'try to do it but maybe not, depending on platform'), and "none" elsewhere.

Indeed, but the `production` flag is a separate one. I.e. if you just type `scons` it does not use LTO, unlike with this PR, which would instead forcibly enable LTO by default unless you specify dev_build, which also has other consequences.

Reply:
I think we can add the `production` flag logic, but I'd do that in a separate PR.

Reply:
Oh, i did not realize 'production' is something else. That's fine then, I'll adjust to what upstream godot does, with no LTO enabled by default.